Upstream status: Fixed in FreeBSD >8.3, >9.0, HEAD.
Obtained from:	 FreeBSD HEAD r237203

Modified: head/sys/dev/fb/fbreg.h
===================================================================
--- head/sys/dev/fb/fbreg.h	2012-06-17 20:45:45 UTC (rev 237202)
+++ head/sys/dev/fb/fbreg.h	2012-06-17 21:02:48 UTC (rev 237203)
@@ -35,9 +35,16 @@
 
 /* some macros */
 #if defined(__amd64__) || defined(__i386__)
-#define bcopy_io(s, d, c)	bcopy((void *)(s), (void *)(d), (c))
-#define bcopy_toio(s, d, c)	bcopy((void *)(s), (void *)(d), (c))
-#define bcopy_fromio(s, d, c)	bcopy((void *)(s), (void *)(d), (c))
+
+static __inline void
+copyw(uint16_t *src, uint16_t *dst, size_t size)
+{
+	while (size--)
+		*dst++ = *src++;
+}
+#define bcopy_io(s, d, c)	copyw((void*)(s), (void*)(d), (c))
+#define bcopy_toio(s, d, c)	copyw((void*)(s), (void*)(d), (c))
+#define bcopy_fromio(s, d, c)	copyw((void*)(s), (void*)(d), (c))
 #define bzero_io(d, c)		bzero((void *)(d), (c))
 #define fill_io(p, d, c)	fill((p), (void *)(d), (c))
 #define fillw_io(p, d, c)	fillw((p), (void *)(d), (c))
